test/localeTest.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+#------------------------------------------------------------------------------
+# Script that tests that the compiler works correctly regardless of the locale
+# setting. As a prerequisite, the following locales must be enabled system-wide:
+# C, tr_TR.utf8, ja_JP.eucjp.
+#
+# Usage:
+#    <script name>.sh <path to solc binary>
+#
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2022 solidity contributors.
+#------------------------------------------------------------------------------
+
+set -eo pipefail
+
+REPO_ROOT=$(cd "$(dirname "$0")/.." && pwd)
+# shellcheck source=scripts/common.sh
+source "${REPO_ROOT}/scripts/common.sh"
+
+solc_binary="$1"
+(( $# == 1 )) || fail "Expected exactly 1 argument."
+
+# This test won't work without some specific locales installed
+locale -a | grep -e "^tr_TR\.utf8$" || fail "Locale 'tr_TR.utf8' not available."
+locale -a | grep -e "^ja_JP\.eucjp$" || fail "Locale 'ja_JP.eucjp' not available."
+locale -a | grep -e "^C$" || fail "Locale 'C' not available."
+locale -a | grep -e "^__invalid_locale__$" && fail "'__invalid_locale__' is not supposed to be a valid locale name."
+
+i="i"
+
+test_code=$(cat <<'EOF'
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity *;
+    library L {}
+EOF
+)
+
+# Whatever locale is set by default.
```

Comment:
Does this test actually fail before the fix?
Can you make it generate some code and check that it is identical for all locales?

Reply:
> Does this test actually fail before the fix?

Yes, see https://github.com/ethereum/solidity/pull/12716#issuecomment-1051261952 and https://github.com/ethereum/solidity/pull/12716#issuecomment-1054096720 above.

> Can you make it generate some code and check that it is identical for all locales?

I assumed that with bad locale it would not compile so this would be enough, but true, we cold write output from all `solc` calls to files (instead of sending it to `/dev/null`) and `diff` it at the end.